src/main/main.c

```diff
@@ -1320,6 +1320,44 @@ void re_thread_leave(void)
 }
 
 
+/**
+ * Attach the current thread to re context
+ */
+int re_thread_attach(struct re *context)
+{
+	struct re *re;
+
+	if (context == NULL) {
+		re_thread_detach();
```

Comment:
Is there a specific use case for auto detach? Maybe its better to return `EINVAL`.

Reply:
No, our code base explicitly calls detach. `EINVAL` is fine.